doc/adr/0012-modules-structure.md

```diff
@@ -0,0 +1,105 @@
+# 12. Modules' structure
+
+Date: 2022-12-28
+
+## Status
+
+Accepted
+
+## Context
+
+The existing structure of tessellation modules was mixed, hence hard to maintain.
+
+- `shared` contains both base classes and extensions, and tessellation schema,
+- `keytool` contains key-pair generator and exposes key-related operations,
+- `wallet` contains operations related to transactions,
+- `kernel` contains part of state channels (hylomorphism's idea),
+- `sdk` contains all the things shared between Global L0 and DAG L1, but not the $DAG itself,
+- `dag-shared` contains all the things shared between Global L0 and DAG L1 related to $DAG,
+- `core` is the Global L0 module (but also contains state channels that must be extracted to be reused in other modules),
+- `dag-l1` is the DAG L1 module.
+
+```mermaid
+graph TD
+    A[Shared]
+    B[Keytool]
+    C[Kernel]
+    D[Wallet]
+    E[SDK]
+    F[DAG Shared]
+    G[Core]
+    H[DAG L1]
+    
+    B --> A
+    C --> A
+    D --> B
+    D -.-> A
+    E --> C
+    E -.-> B
+    E -.-> A
+    E --> F
+    F --> C
+    F -.-> B
+    F -.-> A
+    G --> E
+    G -.-> C
+    G -.-> B
+    G -.-> A
+    G --> F
+    H --> F
```

Comment:
If I understand correctly that dotted line is a transitive dependency. This can be marked as a dotted line as the SDK also requires DAG shared and DAG L1 depends on SDK. Right?

Reply:
actually I removed dotted lines. it was misleading (and not, it wasn't transient but could be)